File: bench/roundtrip/sample.js
// Same user service in JavaScript

function getActiveUsers(users) {
  return users.filter(u => u.active);
}

function topScorers(users, n) {
  return users
    .filter(u => u.active)
    .sort((a, b) => b.score - a.score)
    .slice(0, n);
}

function averageScore(users) {
  if (users.length === 0) return 0;
  const total = users.reduce((acc, u) => acc + u.score, 0);
  return total / users.length;
}

function emailsFor(users) {
  return users.filter(u => u.active).map(u => u.email);
}

File: bench/roundtrip/sample.py
# A realistic Python module — same user service logic

def get_active_users(users):
    return [u for u in users if u["active"]]

def top_scorers(users, n):
    active = [u for u in users if u["active"]]
    return sorted(active, key=lambda u: u["score"], reverse=True)[:n]

def average_score(users):
    if len(users) == 0:
        return 0
    total = sum(u["score"] for u in users)
    return total / len(users)

def emails_for(users):
    return [u["email"] for u in users if u["active"]]
